test/tests/functional/pbs_power_provisioning_cray.py

```diff
@@ -449,15 +500,536 @@ def test_last_used_time(self):
+                            sudo=True)
+        i = 0
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                if n['state'] == 'sleep':
+                    i = i + 1
+        print i
+        if i != 2:
+            self.assertFalse("2 nodes are not in sleep state")
+        self.cleanup_power_on()
+
+    def test_poweroffelgible_false(self):
+        """
+        Test hook wont power off the nodes where
+        poweroff_eligible is set to false
+        """
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.manager(MGR_CMD_SET, NODE,
+                                    {"poweroff_eligible": False}, n['id'])
+        a = {"power_on_off_enable": True,
+             "max_concurrent_nodes": "30", 'node_idle_limit': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+        a = {'freq': 30}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        a = {'enabled': 'True'}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        self.logger.info(
+            "Waiting for 100 secs to make sure no nodes are powered off")
+        time.sleep(100)
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.expect(NODE, {'state': 'free'}, id=n['id'])
+
+    @timeout(900)
+    def test_power_on_nodes(self):
+        """
+        Test when a job is calandered on a vnode which is in sleep state,
+        the node will be powered on and job will run.
+        """
+        self.scheduler.set_sched_config({'strict_ordering': 'True ALL'})
+        self.server.manager(MGR_CMD_SET, NODE, {
+                            "poweroff_eligible": True}, self.names[0])
+        a = {"power_on_off_enable": True, 'node_idle_limit': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+        a = {'freq': 30}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        a = {'enabled': 'True'}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        t = int(time.time())
+        self.logger.info("Waiting for 1 min to poweroff 1st node")
+        time.sleep(60)
+        self.server.log_match(
+            "/opt/cray/capmc/default/bin/capmc node_off", starttime=t)
+        self.server.expect(NODE, {'state': 'sleep'}, id=self.names[0])
+        a = {"node_idle_limit": "1800", 'min_node_down_delay': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+
+        jid = self.submit_job(1000, {'Resource_List.vnode': self.names[0],
+                                     'Resource_List.vntype': 'cray_compute'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
```

Comment:
Unless you have turned scheduling to false, a cycle will run when you submit a job.  There is no need to force one.